src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,28 +18,31 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person with the given phone number used in the displayed person list.\n"
+            + "Parameters: PHONE_NUMBER (must be a valid phone number)\n"
+            + "Example: " + COMMAND_WORD + " 81234567";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private final Phone targetPhone;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Phone targetPhone) {
+        this.targetPhone = targetPhone;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        List<Person> list = model.getAddressBook().getPersonList();
```

Comment:
I see here that it's based on the unfiltered list rather than the filtered list!